client/my-sites/themes/theme-preview.jsx

```diff
@@ -2,79 +2,141 @@
 		showExternal: React.PropTypes.bool,
 		onClose: React.PropTypes.func,
-		primaryButtonLabel: React.PropTypes.string.isRequired,
-		onPrimaryButtonClick: React.PropTypes.func,
-		getPrimaryButtonHref: React.PropTypes.func,
-		secondaryButtonLabel: React.PropTypes.string,
-		onSecondaryButtonClick: React.PropTypes.func,
-		getSecondaryButtonHref: React.PropTypes.func,
-	},
-
-	getDefaultProps() {
-		return {
-			onPrimaryButtonClick: noop,
-			getPrimaryButtonHref: () => null,
-			onSecondaryButtonClick: noop,
-			getSecondaryButtonHref: () => null,
-		};
 	},
 
 	onPrimaryButtonClick() {
-		this.props.onPrimaryButtonClick( this.props.theme );
-		this.props.onClose();
+		const option = this.getPrimaryOption();
+		option.action && option.action( this.props.theme );
+		this.props.hideThemePreview();
 	},
 
 	onSecondaryButtonClick() {
-		this.props.onSecondaryButtonClick( this.props.theme );
-		this.props.onClose();
+		const secondary = this.getSecondaryOption();
+		secondary.action && secondary.action( this.props.theme );
+		this.props.hideThemePreview();
+	},
+
+	getPrimaryOption() {
+		return this.props.themeOptions.primary;
+	},
+
+	getSecondaryOption() {
+		const { isActive } = this.props;
+		return isActive ? null : this.props.themeOptions.secondary;
 	},
 
 	renderSecondaryButton() {
-		if ( ! this.props.secondaryButtonLabel ) {
+		const secondaryButton = this.getSecondaryOption();
+		if ( ! secondaryButton ) {
 			return;
 		}
-		const buttonHref = this.props.getSecondaryButtonHref ? this.props.getSecondaryButtonHref( this.props.theme ) : null;
+		const buttonHref = secondaryButton.getUrl ? secondaryButton.getUrl( this.props.theme ) : null;
 		return (
 			<Button onClick={ this.onSecondaryButtonClick } href={ buttonHref } >
-				{ this.props.secondaryButtonLabel }
+				{ secondaryButton.extendedLabel }
 			</Button>
 		);
 	},
 
 	render() {
-		const previewUrl = getPreviewUrl( this.props.theme );
-		const buttonHref = this.props.getPrimaryButtonHref ? this.props.getPrimaryButtonHref( this.props.theme ) : null;
+		if ( ! this.props.showPreview ) {
+			return null;
+		}
+
+		const primaryOption = this.getPrimaryOption();
+		const buttonHref = primaryOption.getUrl ? primaryOption.getUrl( this.props.theme ) : null;
+		const themeId = this.props.themePreviewInfo.themeId;
 
 		return (
-			<WebPreview
-				showPreview={ this.props.showPreview }
-				showExternal={ this.props.showExternal }
-				showSEO={ false }
-				onClose={ this.props.onClose }
-				previewUrl={ previewUrl }
-				externalUrl={ this.props.theme.demo_uri } >
-				{ this.renderSecondaryButton() }
-				<Button primary onClick={ this.onPrimaryButtonClick } href={ buttonHref } >
-					{ this.props.primaryButtonLabel }
-				</Button>
-			</WebPreview>
+			<div>
+				{ this.props.isJetpack && <QueryTheme themeId={ themeId } siteId="wporg" /> }
+				{ this.props.previewUrl && <WebPreview
+					showPreview={ this.props.showPreview }
+					showExternal={ this.props.showExternal }
```

Comment:
I don't think we're setting this anywhere at all, are we? So we can probably just set it to `false`.